From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Christopher A. Williamson" <contact@chrisaw.io>
Date: Wed, 05 Feb 2026 00:00:00 +0000
Subject: [PATCH 3/4] Remove packages unused in GSI builds

Remove packages that are not useful or not desired in GrapheneOS
GSI builds:

- MusicFX: not included in GrapheneOS
- Auditor: relies on hardware attestation, won't work on GSI
- PdfViewerGOS: can be installed from the app store
- InfoApp: not relevant for GSI builds
- DeviceDiagnostics: cannot properly detect hardware on a GSI
- EasterEgg: novelty app, not needed
- HardeningTestApp: debug/testing app, not for production
- su: root access is managed through Magisk, not AOSP su
- mtectrl/misctrl: Pixel-specific MTE boot control and misc
  partition tools that can interfere with vendor bootloader TEE
  initialization on non-Pixel devices
---
 target/product/base_system.mk      |  4 ----
 target/product/generic/Android.bp  |  1 -
 target/product/handheld_product.mk |  3 ---
 target/product/handheld_system.mk  |  3 ---
 4 files changed, 11 deletions(-)

diff --git a/target/product/generic/Android.bp b/target/product/generic/Android.bp
--- a/target/product/generic/Android.bp
+++ b/target/product/generic/Android.bp
@@ -817,7 +817,6 @@ android_system_image {
                 "MediaProviderLegacy", // base_system
                 "MmsService", // handheld_system
                 "MtpService", // handheld_system
-                "MusicFX", // handheld_system
                 "NetworkStack", // base_system
                 "ONS", // telephony_system
                 "PacProcessor", // handheld_system
diff --git a/target/product/base_system.mk b/target/product/base_system.mk
--- a/target/product/base_system.mk
+++ b/target/product/base_system.mk
@@ -229,8 +229,6 @@ PRODUCT_PACKAGES += \
     mke2fs \
     mkfs.erofs \
     monkey \
-    misctrl \
-    mtectrl \
     ndc \
     netd \
     NetworkStack \
@@ -564,7 +562,6 @@ PRODUCT_PACKAGES_DEBUG := \
     ss \
     start_with_lockagent \
     strace \
-    su \
     sanitizer-status \
     tracepath \
     tracepath6 \
@@ -572,7 +569,6 @@ PRODUCT_PACKAGES += \
     unwind_info \
     unwind_reg_info \
     unwind_symbols \
-    HardeningTestApp \
 
 # The set of packages whose code can be loaded by the system server.
 PRODUCT_SYSTEM_SERVER_APPS += \
diff --git a/target/product/handheld_product.mk b/target/product/handheld_product.mk
--- a/target/product/handheld_product.mk
+++ b/target/product/handheld_product.mk
@@ -23,17 +23,14 @@ $(call inherit-product, $(SRC_TARGET_DIR)/product/media_product.mk)
 # /product packages
 PRODUCT_PACKAGES += \
     AppStore \
-    Auditor \
     Calendar \
     Camera \
     Contacts \
     DeskClock \
     ExactCalculator \
     Gallery2 \
-    InfoApp \
     LatinIME \
     Music \
-    PdfViewerGOS \
     preinstalled-packages-platform-handheld-product.xml \
     SettingsIntelligence \
     ThemePicker \
diff --git a/target/product/handheld_system.mk b/target/product/handheld_system.mk
--- a/target/product/handheld_system.mk
+++ b/target/product/handheld_system.mk
@@ -47,11 +47,9 @@ PRODUCT_PACKAGES += \
     CaptivePortalLogin \
     CertInstaller \
     CredentialManager \
     DeviceAsWebcam \
-    DeviceDiagnostics \
     DocumentsUI \
     DownloadProviderUi \
-    EasterEgg \
     ExternalStorageProvider \
     FusedLocation \
     GmsCompat \
@@ -61,7 +59,6 @@ PRODUCT_PACKAGES += \
     ManagedProvisioning \
     MmsService \
     MtpService \
-    MusicFX \
     NetworkLocation \
     PacProcessor \
     preinstalled-packages-platform-handheld-system.xml \
-- 
2.52.0

